nav2_bt_navigator/behavior_trees/navigate_through_poses_w_replanning_and_recovery.xml

```diff
@@ -14,7 +14,7 @@
         <RateController hz="0.333">
           <RecoveryNode number_of_retries="1" name="ComputePathThroughPoses">
             <ReactiveSequence>
-              <RemovePassedGoals input_goals="{goals}" output_goals="{goals}" radius="0.7"/>
+              <RemovePassedGoals input_goals="{goals}" output_goals="{goals}" radius="0.7" input_waypoint_statuses="{waypoint_statuses}" output_waypoint_statuses="{waypoint_statuses}"/>
```

Comment:
In our Nav2 docs, please update all BT uses of RemovePassedGoals / RemoveInCollision to have these set. I think our migration guide also needs to mention that NavigateThroughPoses navigator now has this waypoint statuses struct in the feedback/result. Also mention how it is managed and what its role is (notifying users about the status of each goal set in the request)